enacts/onset/maproom.py

```diff
@@ -622,73 +676,104 @@ def onset_tile(tz, tx, ty):
         if map_choice == "mean":
-            mymap = onset_dates.onset_delta.mean("T")
-            mymap_max = np.timedelta64(search_days, 'D')
+            map_data = onset_dates.onset_delta.mean("T")
+            map_max = np.timedelta64(search_days, 'D')
         if map_choice == "stddev":
-            mymap = onset_dates.onset_delta.dt.days.std(dim="T", skipna=True)
-            mymap_min = 0
-            mymap_max = int(search_days/3)
+            map_data = onset_dates.onset_delta.dt.days.std(dim="T", skipna=True)
+            map_max = int(search_days/3)
         if map_choice == "pe":
-            mymap = (
+            map_data = (
                 onset_dates.onset_delta.fillna(
                     np.timedelta64(search_days+1, 'D')
-                ) > np.timedelta64(probExcThresh1, 'D')
+                ) > np.timedelta64(prob_exc_thresh1, 'D')
             ).mean("T") * 100
-            mymap_min = 0
-            mymap_max = 100
-            mycolormap = pingrid.CORRELATION_COLORMAP
-    mymap.attrs["colormap"] = mycolormap
-    mymap = mymap.rename(X="lon", Y="lat")
-    mymap.attrs["scale_min"] = mymap_min
-    mymap.attrs["scale_max"] = mymap_max
-    result = pingrid.tile(mymap, tx, ty, tz, clip_shape)
+            map_max = 100
+            colormap = pingrid.CORRELATION_COLORMAP
+        if map_choice == "length_mean":
+            map_data = seasonal_length.mean("T")
+            map_max = np.timedelta64(int(CONFIG["map_text"][map_choice]["map_max"]), 'D')
+        if map_choice == "length_stddev":
+            map_data = seasonal_length.dt.days.std(dim="T", skipna=True)
+            map_max = CONFIG["map_text"][map_choice]["map_max"]
+        if map_choice == "length_pe":
+            map_data = (seasonal_length < np.timedelta64(prob_exc_thresh2, 'D')).mean("T") * 100
+            map_max = 100
+            colormap = pingrid.CORRELATION_COLORMAP
+    map_data.attrs["colormap"] = colormap
+    map_data = map_data.rename(X="lon", Y="lat")
+    map_data.attrs["scale_min"] = map_min
+    map_data.attrs["scale_max"] = map_max
+    result = pingrid.tile(map_data, tx, ty, tz, clip_shape)
 
     return result
 
 
 @APP.callback(
-    Output("colorbar", "children"),
     Output("colorbar", "colorscale"),
     Output("colorbar", "max"),
     Output("colorbar", "tickValues"),
+    Output("colorbar", "unit"),
     Input("search_start_day", "value"),
     Input("search_start_month", "value"),
     Input("searchDays", "value"),
     Input("map_choice", "value")
 )
 def set_colorbar(search_start_day, search_start_month, search_days, map_choice):
-    if map_choice == "pe":
-        return (
-            f"Probabily of onset date to be {search_days} past {search_start_day} {search_start_month}",
-            pingrid.to_dash_colorscale(pingrid.CORRELATION_COLORMAP),
-            int(100),
-            [i for i in range(0, int(100) + 1) if i % 10 == 0],
-        )
+    colorbar = pingrid.to_dash_colorscale(pingrid.RAINBOW_COLORMAP)
```

Comment:
Maybe this line should also be in an if statement since you are redefining it under other if statements?

Reply:
As of now, all the maps use RAINBOW except the pe ones, so colorbar is set to RAINBOW first, and then it's overwritten in the if statements that need to.